.github/workflows/dd-build.yaml

```diff
@@ -0,0 +1,143 @@
+name: Build and Push etcd releases
+
+on:
+  push:
+    # Sequence of patterns matched against refs/heads
+    tags:
+    # Push events on datadog tags
+    - "*-dd*"
+permissions: write-all
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: ["linux/arm64","linux/amd64"]
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Set release version environment variable
+      run: echo RELEASE_VERSION=${GITHUB_REF#refs/tags/} >> $GITHUB_ENV
+      env:
+        GITHUB_REF: ${{ github.ref }}
+    - name: Set target platform environment variable
+      run: echo SANITIZED_TARGET_PLATFORM=${KUBE_BUILD_PLATFORM/\//-} >> $GITHUB_ENV
+      env:
+        KUBE_BUILD_PLATFORM: ${{ matrix.platform }}
+    - name: Build etcd
+      run: REPOSITORY=https://github.com/${{ env.GITHUB_REPOSITORY}}.git ./scripts/build-binary ${{ env.RELEASE_VERSION }}
```

Comment:
There was another script, `./scripts/release.sh`, that I tried to use - it didn't work.

If I supplied a custom tag like `RELEASE_VERSION=v3.5.14-dd.16-aneesh-test`, it still would look for a branch name like `release-3.5`. We seem to have a different branch naming scheme and I didn't want to change the code in `./scripts/release.sh`, so I found this script instead which builds the binaries for us.

If we're interested in changing our branch naming scheme, we can switch to `./scripts/release.sh` in a follow up PR, but for now, this works.